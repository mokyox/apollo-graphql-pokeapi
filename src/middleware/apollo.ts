import { ApolloServer } from "apollo-server-express";
import { typeDefs } from "../graphql/schema/";
import { resolvers } from "../graphql/resolvers";

export const server = new ApolloServer({
  typeDefs,
  resolvers,
  introspection: true,
  playground: true,
});
