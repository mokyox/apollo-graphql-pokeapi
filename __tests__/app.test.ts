import { createTestClient } from "apollo-server-testing";
import { server } from "../src/middleware/apollo";

const { query } = createTestClient(server);
const testCase = {
  query: `
    query Pokemon($id: Int!) {
      getPokemon(id: $id) {
        id
        name
        abilities {
          is_hidden
          ability {
            name
            url
          }
        }
        stats {
          base_stat
          stat {
            name
            url
          }
        }
      }
      getDescription(id: $id) {
        flavor_text
        language {
          name
          url
        }
        version {
          name
          url
        }
      }
    }
    `,
};
describe("app", () => {
  it("should return an object that matches Bulbasaur's data", async () => {
    const response = await query({
      query: testCase.query,
      variables: { id: 1 } || {},
    });

    expect(response).toMatchSnapshot();
  });
  it("should return an object that matches Dartrix's data", async () => {
    const response = await query({
      query: testCase.query,
      variables: { id: 723 } || {},
    });

    expect(response).toMatchSnapshot();
  });
  it("should return an object that matches Zacian's data", async () => {
    const response = await query({
      query: testCase.query,
      variables: { id: 888 } || {},
    });

    expect(response).toMatchSnapshot();
  });
});
