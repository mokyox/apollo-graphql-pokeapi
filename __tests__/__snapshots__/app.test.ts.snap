// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should return an object that matches Bulbasaur's data 1`] = `
Object {
  "data": Object {
    "getDescription": Object {
      "flavor_text": "Bulbasaur can be seen napping in bright sunlight.
There is a seed on its back. By soaking up the sun’s rays,
the seed grows progressively larger.",
      "language": Object {
        "name": "en",
        "url": "https://pokeapi.co/api/v2/language/9/",
      },
      "version": Object {
        "name": "alpha-sapphire",
        "url": "https://pokeapi.co/api/v2/version/26/",
      },
    },
    "getPokemon": Object {
      "abilities": Array [
        Object {
          "ability": Object {
            "name": "overgrow",
            "url": "https://pokeapi.co/api/v2/ability/65/",
          },
          "is_hidden": false,
        },
        Object {
          "ability": Object {
            "name": "chlorophyll",
            "url": "https://pokeapi.co/api/v2/ability/34/",
          },
          "is_hidden": true,
        },
      ],
      "id": 1,
      "name": "bulbasaur",
      "stats": Array [
        Object {
          "base_stat": 45,
          "stat": Object {
            "name": "hp",
            "url": "https://pokeapi.co/api/v2/stat/1/",
          },
        },
        Object {
          "base_stat": 49,
          "stat": Object {
            "name": "attack",
            "url": "https://pokeapi.co/api/v2/stat/2/",
          },
        },
        Object {
          "base_stat": 49,
          "stat": Object {
            "name": "defense",
            "url": "https://pokeapi.co/api/v2/stat/3/",
          },
        },
        Object {
          "base_stat": 65,
          "stat": Object {
            "name": "special-attack",
            "url": "https://pokeapi.co/api/v2/stat/4/",
          },
        },
        Object {
          "base_stat": 65,
          "stat": Object {
            "name": "special-defense",
            "url": "https://pokeapi.co/api/v2/stat/5/",
          },
        },
        Object {
          "base_stat": 45,
          "stat": Object {
            "name": "speed",
            "url": "https://pokeapi.co/api/v2/stat/6/",
          },
        },
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`app should return an object that matches Dartrix's data 1`] = `
Object {
  "data": Object {
    "getDescription": Object {
      "flavor_text": "It throws one knifelike feather after another at its
enemies, and each one precisely strikes a weak
point. These feathers are known as blade quills.",
      "language": Object {
        "name": "en",
        "url": "https://pokeapi.co/api/v2/language/9/",
      },
      "version": Object {
        "name": "sword",
        "url": "https://pokeapi.co/api/v2/version/33/",
      },
    },
    "getPokemon": Object {
      "abilities": Array [
        Object {
          "ability": Object {
            "name": "overgrow",
            "url": "https://pokeapi.co/api/v2/ability/65/",
          },
          "is_hidden": false,
        },
        Object {
          "ability": Object {
            "name": "long-reach",
            "url": "https://pokeapi.co/api/v2/ability/203/",
          },
          "is_hidden": true,
        },
      ],
      "id": 723,
      "name": "dartrix",
      "stats": Array [
        Object {
          "base_stat": 78,
          "stat": Object {
            "name": "hp",
            "url": "https://pokeapi.co/api/v2/stat/1/",
          },
        },
        Object {
          "base_stat": 75,
          "stat": Object {
            "name": "attack",
            "url": "https://pokeapi.co/api/v2/stat/2/",
          },
        },
        Object {
          "base_stat": 75,
          "stat": Object {
            "name": "defense",
            "url": "https://pokeapi.co/api/v2/stat/3/",
          },
        },
        Object {
          "base_stat": 70,
          "stat": Object {
            "name": "special-attack",
            "url": "https://pokeapi.co/api/v2/stat/4/",
          },
        },
        Object {
          "base_stat": 70,
          "stat": Object {
            "name": "special-defense",
            "url": "https://pokeapi.co/api/v2/stat/5/",
          },
        },
        Object {
          "base_stat": 52,
          "stat": Object {
            "name": "speed",
            "url": "https://pokeapi.co/api/v2/stat/6/",
          },
        },
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`app should return an object that matches Zacian's data 1`] = `
Object {
  "data": Object {
    "getDescription": Object {
      "flavor_text": "Known as a legendary hero, this Pokémon
absorbs metal particles, transforming them
into a weapon it uses to battle.",
      "language": Object {
        "name": "en",
        "url": "https://pokeapi.co/api/v2/language/9/",
      },
      "version": Object {
        "name": "sword",
        "url": "https://pokeapi.co/api/v2/version/33/",
      },
    },
    "getPokemon": Object {
      "abilities": Array [
        Object {
          "ability": Object {
            "name": "intrepid-sword",
            "url": "https://pokeapi.co/api/v2/ability/234/",
          },
          "is_hidden": false,
        },
      ],
      "id": 888,
      "name": "zacian-hero",
      "stats": Array [
        Object {
          "base_stat": 92,
          "stat": Object {
            "name": "hp",
            "url": "https://pokeapi.co/api/v2/stat/1/",
          },
        },
        Object {
          "base_stat": 130,
          "stat": Object {
            "name": "attack",
            "url": "https://pokeapi.co/api/v2/stat/2/",
          },
        },
        Object {
          "base_stat": 115,
          "stat": Object {
            "name": "defense",
            "url": "https://pokeapi.co/api/v2/stat/3/",
          },
        },
        Object {
          "base_stat": 80,
          "stat": Object {
            "name": "special-attack",
            "url": "https://pokeapi.co/api/v2/stat/4/",
          },
        },
        Object {
          "base_stat": 115,
          "stat": Object {
            "name": "special-defense",
            "url": "https://pokeapi.co/api/v2/stat/5/",
          },
        },
        Object {
          "base_stat": 138,
          "stat": Object {
            "name": "speed",
            "url": "https://pokeapi.co/api/v2/stat/6/",
          },
        },
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
